### DISCUSSION TITLE
how to specify which foreign key to use when you are referencing a table with multiple foreign keys

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#809] [Questions] [answered]
I have two different ids that relate to one table, how do i specify which foreign key i am using?

 const { data: dealData, error: dealError } = await supabase
        .from("connections")
        .select(`*,
        deals(*
          ),
          profiles(*
            )`)
        .match({ recipient: userId })

returns:  message: 'More than one relationship was found for connections and profiles'

**Comment by steve-chavez** [comment, marked as answer]
Hey @pearlslugs,

Try specifying the fk(or column that belongs to the fk) like:

```js
const { data: dealData, error: dealError } = await supabase
.from("connections")
.select("*, deals!fk_or_col_1(* ), profiles!fk_or_col2(* )")
.match({ recipient: userId })
```

For more details, check: https://postgrest.org/en/v7.0.0/api.html#embedding-disambiguation

**Reply by ghost** [reply]
Thank you :)

this is the correct code with !sender being the column in connections that has a foreign key to profiles, for anyone who comes across the same problem, here the solution is, clear as day :)


```
const { data: dealData, error: dealError } = await supabase
        .from("connections")
        .select("*, deals(* ),  profiles!sender(* )")
        .match({ recipient: userId });
```
